Blueprint.Api/Services/ScenarioEventService.cs

```diff
@@ -141,6 +144,128 @@ public async Task<IEnumerable<ViewModels.ScenarioEvent>> CreateAsync(ViewModels.
             return  _mapper.Map<IEnumerable<ViewModels.ScenarioEvent>>(scenarioEventEnitities);
         }
 
+        public async Task<IEnumerable<ViewModels.ScenarioEvent>> CreateFromInjectsAsync(CreateFromInjectsForm createFromInjectsForm, CancellationToken ct)
+        {
+            var userId = _user.GetId();
+            // user must be a Content Developer or a MSEL owner
+            if (!(await _authorizationService.AuthorizeAsync(_user, null, new ContentDeveloperRequirement())).Succeeded &&
+                !(await MselOwnerRequirement.IsMet(userId, createFromInjectsForm.MselId, _context)))
+                throw new ForbiddenException();
+
+            // get the MSEL
+            var msel = await _context.Msels
+                .Include(m => m.DataFields)
+                .SingleOrDefaultAsync(x => x.Id == createFromInjectsForm.MselId);
+            if (msel == null)
+                throw new EntityNotFoundException<MselEntity>("Msel " + createFromInjectsForm.MselId.ToString() + " was not found.");
+
+            // get the inject type
+            var injectType = await _context.InjectTypes
+                .Include(m => m.DataFields)
+                .ThenInclude(n => n.DataOptions)
+                .AsSplitQuery()
+                .SingleOrDefaultAsync(m => m.Id == createFromInjectsForm.InjectTypeId);
+            if (injectType == null)
+                throw new EntityNotFoundException<InjectTypeEntity>("Inject Type " + createFromInjectsForm.InjectTypeId.ToString() + " was not found.");
+
+            // start a transaction, because we may also update DataValues and other scenario events
+            await _context.Database.BeginTransactionAsync();
+            var dateCreated = DateTime.UtcNow;
+            // create a data field dictionary and create any missing data fields
+            var dataFieldDictionary = new Dictionary<Guid, Guid>();
+            var displayOrder = msel.DataFields.Count > 0
+                ? msel.DataFields.Max(m => m.DisplayOrder)
+                : 0;
+            foreach (var injectDataField in injectType.DataFields)
+            {
+                var showIt = injectDataField.DataType != DataFieldType.Html;
+                var mselDataFieldEntity = msel.DataFields.SingleOrDefault(m => m.Name == injectDataField.Name && m.DataType == injectDataField.DataType);
+                if (mselDataFieldEntity == null)
+                {
+                    mselDataFieldEntity = new DataFieldEntity()
+                    {
+                        Id = Guid.NewGuid(),
+                        MselId = msel.Id,
+                        Name = injectDataField.Name,
+                        DataType = injectDataField.DataType,
+                        DisplayOrder = ++displayOrder,
+                        DateCreated = dateCreated,
+                        CreatedBy = userId,
+                        IsChosenFromList = injectDataField.IsChosenFromList,
+                        OnScenarioEventList = showIt,
```

Comment:
Wouldn't you want to get this from the inject DataField if it is there and its not an html type?